airbyte-integrations/connectors/source-linkedin-ads/source_linkedin_ads/streams.py

```diff
@@ -389,3 +389,73 @@ def get_updated_state(self, current_stream_state: MutableMapping[str, Any], late
+    ) -> Mapping[str, Any]:
+        headers = super().request_headers(stream_state, stream_slice, next_page_token)
+        headers.update({"X-Restli-Protocol-Version": "2.0.0", "Linkedin-version": "202401"})
+        return headers
+
+    def request_params(
+        self,
+        stream_state: Mapping[str, Any],
+        stream_slice: Mapping[str, Any] = None,
+        next_page_token: Mapping[str, Any] = None,
+    ) -> MutableMapping[str, Any]:
+        params = super().request_params(stream_state, stream_slice, next_page_token)
+        params["q"] = "owner"
+        params["owner"] = f"(sponsoredAccount:urn%3Ali%3AsponsoredAccount%3A{stream_slice.get('account_id')})"
+        params["leadType"] = "(leadType:SPONSORED)"
+        return urlencode(params, safe="():,%")
+
+    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]) -> Mapping[str, Any]:
+        current_stream_state = (
+            {self.cursor_field: pendulum.parse(self.config.get("start_date")).format("x")}
+            if not current_stream_state
+            else current_stream_state
+        )
+        return {self.cursor_field: max(latest_record.get(self.cursor_field), int(current_stream_state.get(self.cursor_field)))}
+
+class Forms(LinkedInAdsStreamSlicing):
+    """
+        Get forms data using `account_id` slicing.
+    https://learn.microsoft.com/en-us/linkedin/marketing/lead-sync/leadsync?view=li-lms-2023-11&tabs=http"""
+
+    endpoint = "leadForms"
+    cursor_field = "lastModified"
+
+    def request_headers(
+        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> Mapping[str, Any]:
+        headers = super().request_headers(stream_state, stream_slice, next_page_token)
+        headers.update({"X-Restli-Protocol-Version": "2.0.0", "Linkedin-version": "202401"})
+        return headers
+
+    def request_params(
+        self,
+        stream_state: Mapping[str, Any],
+        stream_slice: Mapping[str, Any] = None,
+        next_page_token: Mapping[str, Any] = None,
+    ) -> MutableMapping[str, Any]:
+        params = super().request_params(stream_state, stream_slice, next_page_token)
+        params["q"] = "owner"
+        params["owner"] = f"(sponsoredAccount:urn%3Ali%3AsponsoredAccount%3A{stream_slice.get('account_id')})"
+        return urlencode(params, safe="():,%")
+    
+    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]) -> Mapping[str, Any]:
+        current_stream_state = (
+            {self.cursor_field: pendulum.parse(self.config.get("start_date")).format("x")}
+            if not current_stream_state
+            else current_stream_state
+        )
+        return {self.cursor_field: max(latest_record.get(self.cursor_field), int(current_stream_state.get(self.cursor_field)))}
```

Comment:
Both Leads and Forms have similar functions, maybe create a base stream class and reused?